privacy/bolton/losses.py

```diff
@@ -0,0 +1,319 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Loss functions for bolton method"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import tensorflow as tf
+from tensorflow.python.keras import losses
+from tensorflow.python.keras.utils import losses_utils
+from tensorflow.python.framework import ops as _ops
+from tensorflow.python.keras.regularizers import L1L2
+from tensorflow.python.platform import tf_logging as logging
+
+
+class StrongConvexMixin:
+  """
+  Strong Convex Mixin base class for any loss function that will be used with
+  Bolton model. Subclasses must be strongly convex and implement the
+  associated constants. They must also conform to the requirements of tf losses
+  (see super class).
+
+  For more details on the strong convexity requirements, see:
+  Bolt-on Differential Privacy for Scalable Stochastic Gradient
+  Descent-based Analytics by Xi Wu et. al.
+  """
+
+  def radius(self):
+    """Radius, R, of the hypothesis space W.
+    W is a convex set that forms the hypothesis space.
+
+    Returns: R
+
+    """
+    raise NotImplementedError("Radius not implemented for StrongConvex Loss"
+                              "function: %s" % str(self.__class__.__name__))
+
+  def gamma(self):
+    """ Strongly convexity, gamma
+
+    Returns: gamma
+
+    """
+    raise NotImplementedError("Gamma not implemented for StrongConvex Loss"
+                              "function: %s" % str(self.__class__.__name__))
+
+  def beta(self, class_weight):
+    """Smoothness, beta
+
+    Args:
+      class_weight: the class weights as scalar or 1d tensor, where its
+                    dimensionality is equal to the number of outputs.
+
+    Returns: Beta
+
+    """
+    raise NotImplementedError("Beta not implemented for StrongConvex Loss"
+                              "function: %s" % str(self.__class__.__name__))
+
+  def lipchitz_constant(self, class_weight):
+    """Lipchitz constant, L
+
+    Args:
+      class_weight: class weights used
+
+    Returns: L
+
+    """
+    raise NotImplementedError("lipchitz constant not implemented for "
+                              "StrongConvex Loss"
+                              "function: %s" % str(self.__class__.__name__))
+
+  def kernel_regularizer(self):
+    """returns the kernel_regularizer to be used. Any subclass should override
+      this method if they want a kernel_regularizer (if required for
+      the loss function to be StronglyConvex
+
+    :return: None or kernel_regularizer layer
```

Comment:
":return:" -> "Returns"

And actually start of the comments already mention returns, you should consider consolidate these two